File: backend/src/main/java/protasks/backend/Board/BoardUsersPermRel.java
package protasks.backend.Board;

import com.fasterxml.jackson.annotation.JsonView;
import protasks.backend.Rol.Rol;
import protasks.backend.user.User;

import javax.persistence.*;
import java.util.Objects;

@Entity
@Table(name = "boards_users_rel")
public class BoardUsersPermRel {
    public interface BoardBasicInfo{}
    public interface UserBasicInfo{}

    @JsonView(BoardUsersPermRel.class)
    @EmbeddedId
    private BoardUsersPermId id;

    @JsonView(UserBasicInfo.class)
    @ManyToOne
    @MapsId("boardId")
    private Board board;

    @JsonView(BoardBasicInfo.class)
    @ManyToOne
    @MapsId("userId")
    private User user;

    @JsonView(BoardUsersPermRel.class)
    @Column(name = "rol")
    private Rol rol ;

    public BoardUsersPermRel() {}

    public BoardUsersPermRel(Board board, User user,Rol rol) {
        this.board = board;
        this.user = user;
        this.rol=rol;
        this.id = new BoardUsersPermId(board.getId(), user.getId());
    }

    public BoardUsersPermId getId() {
        return id;
    }

    public void setId(BoardUsersPermId id) {
        this.id = id;
    }

    public Board getBoard() {
        return board;
    }

    public void setBoard(Board board) {
        this.board = board;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Rol getRol() {
        return rol;
    }

    public void setRol(Rol rol) {
        this.rol = rol;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;

        if (o == null || getClass() != o.getClass())
            return false;

        BoardUsersPermRel that = (BoardUsersPermRel) o;
        return Objects.equals(user, that.user) &&
                Objects.equals(board, that.board);
    }

    @Override
    public int hashCode() {
        return Objects.hash(board, user);
    }
}

File: app/src/main/java/com/example/protasks/models/BoardUsersPermRel.kt
package com.example.protasks.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.util.*

class BoardUsersPermRel {

    @Expose
    @SerializedName("Id")
    private var id: BoardUsersPermId? = null

    @Expose
    @SerializedName("board")
    private var board: Board? = null

    @Expose
    @SerializedName("user")
    private var user: User? = null

    @Expose
    @SerializedName("rol")
    private var rol: Rol? = null

    fun BoardUsersPermRel() {}

    fun BoardUsersPermRel(
        board: Board,
        user: User,
        rol: Rol?
    ) {
        this.board = board
        this.user = user
        this.rol = rol
        id = BoardUsersPermId(board.getId(), user.getId())
    }

    fun getId(): BoardUsersPermId? {
        return id
    }

    fun setId(id: BoardUsersPermId?) {
        this.id = id
    }

    fun getBoard(): Board? {
        return board
    }

    fun setBoard(board: Board?) {
        this.board = board
    }

    fun getUser(): User? {
        return user
    }

    fun setUser(user: User?) {
        this.user = user
    }

    fun getRol(): Rol? {
        return rol
    }

    fun setRol(rol: Rol?) {
        this.rol = rol
    }

}